pulser/sequence.py

```diff
@@ -243,7 +246,7 @@ def current_phase_ref(self, qubit: QubitId,
 
     def declare_channel(self, name: str, channel_id: str,
                         initial_target: Optional[
-                            Union[Iterable[Union[QubitId, Parametrized]],
+                            Union[Set[Union[QubitId, Parametrized]],
```

Comment:
Even though what's stored in `_TimeSlot.targets` is a `set`, we can (and want to) accept an `Iterable` here